ldap/src/integration-test/java/org/springframework/security/ldap/SpringLdapFalseAuthTest.java

```diff
@@ -0,0 +1,37 @@
+package org.springframework.security.ldap;
+
+import com.unboundid.ldap.listener.InMemoryDirectoryServer;
+import com.unboundid.ldap.sdk.LDAPException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.ldap.server.ApacheDSContainer;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = ApacheDsContainerConfig.class)
+public class SpringLdapFalseAuthTest {
+
+	@Test
+	public void testApacheDSContainerInvalidLdifFile(){
+
+		//TODO - get exception here in case of invalid root base
+		new ApacheDSContainer("dc=springframework,dc=org",
+				"classpath:missing-file.ldif");
+	}
+
+	@Test(expected = LDAPException.class)
+	public void testInMemoryDirectoryServerInvalidLdifFile(){
+		//TODO - get exception here in case of invalid root base
+		try {
+			new InMemoryDirectoryServer("dc=springframework,dc=org",
+					"classpath:missing-file.ldif");
+		} catch (LDAPException e) {
```

Comment:
An `IllegalArgumentException` is more suitable in this case.
`LDAPException` should only be used when a problem occurs while performing LDAP-related processing.
In this case, we haven't started the LDAP-related processing yet, so we should use `IllegalArgumentException`.

Reply:
The ```InMemoryDirectoryServer``` throw `LDAPException` I can wrap it but I think it's not a good practice.
`  public InMemoryDirectoryServer(final String... baseDNs)
         throws LDAPException`